File: client/src/types/order.ts
export interface Order {
  orderId: number;
  userId: number | null;
  status: string;
  totalCents: number;
  paymentCompletedAt: string | null;
  orderItems: OrderItem[];
}

export interface OrderItem {
  orderItemId: number;
  productId: number;
  productName: string;
  unitPriceCents: number;
  quantity: number;
  totalCents: number;
}

File: server/API/Models/DsfTables/Order.cs
using API.Database;

namespace API.Models.DsfTables;

[DbTable("dsf.[order]")]
public class Order
{
    [DbPrimaryKey] public int OrderId { get; set; }
    [DbColumn] public int? UserId { get; set; } // guest mode for demo
    [DbColumn] public string? StripeSessionId { get; set; }
    [DbColumn] public string? StripePaymentIntentId { get; set; }
    [DbColumn] public string Status { get; set; } = "Pending";
    [DbColumn] public int TotalCents { get; set; }
    [DbColumn] public string? Email { get; set; }
    [DbColumn] public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    [DbColumn] public DateTime? UpdatedAt { get; set; }
    [DbColumn] public DateTime? PaymentCompletedAt { get; set; }
}